File: client/src/main/java/net/treset/minecraftlauncher/util/ui/sort/InstanceDetailsNameComparator.java
package net.treset.minecraftlauncher.util.ui.sort;

import net.treset.minecraftlauncher.LauncherApplication;
import net.treset.minecraftlauncher.data.InstanceData;

import java.util.Comparator;

public class InstanceDetailsNameComparator implements Comparator<InstanceData> {
    @Override
    public int compare(InstanceData e1, InstanceData e2) {
        return e1.getInstance().getKey().getName().compareTo(e2.getInstance().getKey().getName());
    }

    @Override
    public String toString() {
        if(LauncherApplication.stringLocalizer == null) {
            return "sort.name";
        }
        return LauncherApplication.stringLocalizer.get("sort.name");
    }
}

File: client/composeApp/src/commonMain/kotlin/net/treset/treelauncher/backend/util/sort/InstanceDetailsNameComparator.kt
package net.treset.treelauncher.backend.util.sort

import net.treset.treelauncher.backend.data.InstanceData
import net.treset.treelauncher.localization.strings

class InstanceDetailsNameComparator : Comparator<InstanceData> {
    override fun compare(e1: InstanceData, e2: InstanceData): Int {
        return e1.instance.first.name.lowercase().compareTo(e2.instance.first.name.lowercase())
    }

    override fun toString(): String = strings().sortBox.sort.name()
}
